src/Serilog/Events/StructureValue.cs

```diff
@@ -37,6 +37,12 @@ public StructureValue(IEnumerable<LogEventProperty> properties, string? typeTag
         _properties = properties.ToArray();
     }
 
+    internal StructureValue(LogEventProperty[] properties, string? typeTag)
+    {
+        TypeTag = typeTag;
+        _properties = properties.ToArray();
```

Comment:
Is the ```ToArray()``` needed?

Reply:
Nice catch, thanks! Missed this when I `Ctrl-Z`-ed out the hacky version of the change and replaced it with the backwards-compatible one 😅  --- will re-benchmark!